### PR TITLE
Refactor BadBlockReason enum out of BadBlockCause

### PR DESCRIPTION
**Pull request by usmansaleem**
### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Most advanced CI tests are deferred until PR approval, but you could:

- [x] locally run all unit tests via: `./gradlew build`
- [x] locally run all acceptance tests via: `./gradlew acceptanceTest`
- [x] locally run all integration tests via: `./gradlew integrationTest`
- [x] locally run all reference tests via: `./gradlew ethereum:referenceTests:referenceTests`

## PR description
BadBlockReason was an non-static inner enum in BadBlockCause and it was used in static context. This caused one of our custom errorprone checks to not pick up the "method param not final argument" correctly. Refactoring it out as an static top level class. Now all the checks are applied correctly.

## Fixed Issue(s)

#6685

### CODE REVIEW

**Review by macfarla — APPROVED** [review]
LGTM

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockReason.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.chain;
+
+public enum BadBlockReason {
+  // Standard spec-related validation failures
+  SPEC_VALIDATION_FAILURE,
+  // When an unexpected exception occurs during block processing
+  EXCEPTIONAL_BLOCK_PROCESSING,
```
Comment (mbaxter) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockReason.java:21]: (optional) While you're here - this value is not used anywhere and can be removed.  If you want to leave it for now, I can clean this up in another PR.

**Comment by mbaxter** [comment]
Does the custom errorprone check need to be updated to avoid this in the future?  Or do we need another rule that enforces static inner classes?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockReason.java (hunk 1 above)
Comment (usmansaleem) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockReason.java]: done

**Comment by usmansaleem** [comment]
@mbaxter I am extracting out Besu errorprone in its own repo. https://github.com/hyperledger/besu-errorprone-checks/ . I tried to reproduce this error via test case, however, I wasn't able to reproduce it: https://github.com/hyperledger/besu-errorprone-checks/blob/main/src/test/resources/org/hyperledger/errorpronechecks/ConstructorParameterMustBeFinalPositiveCases.java

I suspect in addition of slightly buggy custom errorprone check, we have some other complexity involved here as well (perhaps dagger compiler?).

**Pull request merged**